### PR TITLE
Fixed bug that prohibited Admin users from creating new blogging category

### PR DESCRIPTION
**Pull request by trowbrsa**
There was a bug that did not allow admin users to create new blogging categories. This was because view for adding a new category is managed by the 'new_or_edit' function in the admin categories controller.  Previously, this method required categories to have an ID. Since new categories will not have an id until they are saved later on in the method, the page was erroring out. I updated the method by creating a wrapper for new categories that are instantiated. Let me know if you have any questions. Thanks!

### CODE REVIEW

File: features/create_categories.feature [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    And  I press "Save"
+    Then I should see "Category was successfully saved."
+
+    Scenario: Edit a Category
+      When I go to the edit page for "applesauce"
+      And I fill in "Description" with "Tasty"
+      And I press "Save"
+      And I should see "Category was successfully saved."
+
+      Scenario: Cancel editing a category
```
Comment (kariabancroft) [inline features/create_categories.feature]: Looks like your indentation in these scenarios is a little bit off

File: features/create_categories.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    And I fill in "Permalink" with "www.link.com"
+    And I fill in "Description" with "A compelling description"
+    And  I press "Save"
+    Then I should see "Category was successfully saved."
+
+    Scenario: Edit a Category
+      When I go to the edit page for "applesauce"
+      And I fill in "Description" with "Tasty"
+      And I press "Save"
+      And I should see "Category was successfully saved."
```
Comment (kariabancroft) [inline features/create_categories.feature]: Do you also want to check to see if you can see `applesauce`?

File: spec/controllers/admin/categories_controller_spec.rb [hunk 3]
```diff
@@ -11,9 +11,14 @@
     request.session = { :user => henri.id }
   end
 
+  it "successfully loads" do
+    get :new
+    expect(response.status).to eq 200
```
Comment (kariabancroft) [inline spec/controllers/admin/categories_controller_spec.rb:16]: Do you also want to check that it has rendered the correct template?

File: features/create_categories.feature (hunk 2 above)
Comment (trowbrsa) [inline features/create_categories.feature]: This is a great idea, though I tried many iterations of the test and could not find a way to test if the word 'applesauce' was on the page.

File: spec/controllers/admin/categories_controller_spec.rb (hunk 3 above)
Comment (trowbrsa) [inline spec/controllers/admin/categories_controller_spec.rb:16]: Great idea, thanks - done!

**Comment by trowbrsa** [comment]
I've implemented your suggested edits regarding indentation and updating the test to ensure that the correct view is rendered in Rspec. I have not found a great way to test your 2nd suggestion, checking to see if the name field in Categories is in fact the name field for which I want to make the edits.

**Comment by kariabancroft** [comment]
Great! Good to close

**Pull request closed without merging**